src/main/java/com/alvarium/sign/Ed25519Provider.java

```diff
@@ -0,0 +1,40 @@
+package com.alvarium.sign;
+
+import java.security.GeneralSecurityException;
+
+import com.alvarium.utils.Encoder;
+import com.google.crypto.tink.subtle.Ed25519Sign;
+import com.google.crypto.tink.subtle.Ed25519Verify;
+
+public class Ed25519Provider implements SignProvider {
+
+  public String sign(byte[] key, byte[] data) throws SignException {
+    
+    Ed25519Sign signer;
+
+    try {
+      signer = new Ed25519Sign(key);
+    } catch(GeneralSecurityException e) {
+      throw new SignTypeException("SHA512 not supported");
+    }
+
+    try {
+      final byte[] signed = signer.sign(data);
+      final String signedString = Encoder.bytesToHex(signed);
+      return signedString;
+    } catch(GeneralSecurityException e) {
+      throw new SignException("Invalid key or data format");
+    }
+  }
+
+  public boolean verify(byte[] key, byte[] content, byte[] signed) {
+    final Ed25519Verify verifier = new Ed25519Verify(key);
+    try {
+      verifier.verify(signed, content);
+      return true;
+    } catch(GeneralSecurityException e) {
+      return false;
```

Comment:
This exception is just thrown away -- I think we'd want to know WHY this failed rather than just returning false.

Reply:
Unfortunately, the library being used to call `verify` throws a `GeneralSecurityException` when the verification fails, which is mainly when something goes wrong (scenario where I'd normally throw an exception) or the verification failed (scenario where I'd return false) but both exceptions are identifiable from each other by the message only. 

I do agree that we would need to know why this failed, should I change the `SignProvider` interface to follow this pattern where it would throw exceptions when the verification doesn't pass ? or maybe check the message and act accordingly but I can't decide if that's a good idea though.